### PR TITLE
add k8s namespace get and create functionality

### PR DESCRIPTION
**Pull request by onorua**
### What does this PR do?

Add management of namespaces. With it you can create and get either all namespaces or particular namespace. 
### What issues does this PR fix or reference?

Missing or limited k8s functionality 
### Previous Behavior

Had only labels management
### New Behavior

has labels + namespaces (which is entry point for all the rest kuberenetes objects)
### Tests written?
- [ ] Yes
- [x] No

This is my first PR to salt, so I decided to get feedback for modules
before I go with states and continue to work on other parts such as:
service and replication controller.

### CODE REVIEW

**Comment by onorua** [comment]
Weird error with no explanation from your Jenkins

**Comment by onorua** [comment]
yes, just verify, in case it is generally okay, I'm almost ready to push secrets management, need some testings

**Comment by jtand** [comment]
Go Go Jenkins!

**Comment by jtand** [comment]
Go Go Jenkins

File: salt/modules/k8s.py [hunk 1]
```diff
@@ -130,6 +200,27 @@ def get_labels(node=None, apiserver_url=None):
     return {"labels": ret}
 
 
+def get_namespaces(namespace="", apiserver_url=None):
+    '''
+    Get k8s namespaces
+
```
Comment (rallytime) [inline salt/modules/k8s.py]: Can you please add a `.. versionadded:: Carbon` directive here? As well as some docs explaining what the kwargs are?

File: salt/modules/k8s.py [hunk 2]
```diff
@@ -261,3 +352,39 @@ def label_folder_absent(
         ret['comment'] = "Label folder {0} absent".format(folder)
 
     return ret
+
+def namespace_present(
+        name,
+        apiserver_url=None):
+    '''
+    Create k8s namespace 
+
```
Comment (rallytime) [inline salt/modules/k8s.py]: A versionadded tag and docs for the arg/kwarg should be added here as well.

File: salt/modules/k8s.py [hunk 3]
```diff
@@ -261,3 +352,39 @@ def label_folder_absent(
         ret['comment'] = "Label folder {0} absent".format(folder)
 
     return ret
+
+def namespace_present(
```
Comment (rallytime) [inline salt/modules/k8s.py]: Can you put these all up on one line?

**Comment by rallytime** [comment]
Thanks for this addition, @onorua! I had some comments for a couple of small things, plus there are some [pylint errors](https://jenkins.saltstack.com/job/salt-pr-lint-n/14118/violations/file/salt/modules/k8s.py/) that need to be addressed. (The other test failures are not related.)

Once that is cleaned up we can get this in!

**Comment by onorua** [comment]
Just a question, I'm ready to push secrets management from salt, it makes a lot of original pull request, such as using utils.http instead of requests (this was in backlog, so I decided to implement it), and some refactoring such as status 409 from server. 
The question is - should I do just a rebase or close this one?

**Comment by rallytime** [comment]
@onorua Whatever is easiest for you is just fine. :)

**Comment by onorua** [comment]
JFYI, I'm stuck with test automating, for some reason it is missing `__salt__`:
```
./tests/runtests.py -n unit.modules.k8s_test
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 * Transplanting configuration files to '/tmp/salt-tests-tmpdir/config'
 * Current Directory: /opt/salt
 * Test suite is running under PID 17982
 * Logging tests on /tmp/salt-runtests.log
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Starting unit.modules.k8s_test Tests
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
EE
======================================================================
ERROR: test_get_namespaces (unit.modules.k8s_test.TestK8SNamespace)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/salt/tests/unit/modules/k8s_test.py", line 20, in test_get_namespaces
    res = k8s.get_namespaces()
  File "/opt/salt/salt/modules/k8s.py", line 322, in get_namespaces
    apiserver_url = _guess_apiserver(apiserver_url)
  File "/opt/salt/salt/modules/k8s.py", line 95, in _guess_apiserver
    elif __salt__['config.get']('k8s:master'):
NameError: global name '__salt__' is not defined

======================================================================
ERROR: test_get_one_namespace (unit.modules.k8s_test.TestK8SNamespace)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/salt/tests/unit/modules/k8s_test.py", line 26, in test_get_one_namespace
    res = k8s.get_namespaces("default")
  File "/opt/salt/salt/modules/k8s.py", line 322, in get_namespaces
    apiserver_url = _guess_apiserver(apiserver_url)
  File "/opt/salt/salt/modules/k8s.py", line 95, in _guess_apiserver
    elif __salt__['config.get']('k8s:master'):
NameError: global name '__salt__' is not defined

----------------------------------------------------------------------
Ran 2 tests in 0.003s

FAILED (errors=2)
```

What am I doing wrong? I really want to keep good code quality and I think it must be tested, but so far I'm lost :( Any help will be highly appreciated!

**Comment by rallytime** [comment]
@onorua Unit tests rely heavily on mocking, and don't come with running salt daemons, which populate `__salt__` dict. In a unit test, you need to mock the `__salt__` dictionary. 

Here are some useful docs for test writing, if you haven't seen them yet:
- [Writing Unit Tests](https://docs.saltstack.com/en/latest/topics/development/tests/unit.html)
- [Salt's Test Suite: An Introduction](https://docs.saltstack.com/en/latest/topics/tutorials/writing_tests.html)

An you can see an example of a unit test file mocking various salt dictionaries in the [archive module tests](https://github.com/saltstack/salt/blob/develop/tests/unit/modules/archive_test.py).

**Comment by cachedout** [comment]
Go Go Jenkins!

**Pull request closed without merging**